src/Microsoft.AspNet.Mvc.DataAnnotations/ValidationAttributeAdapterOfTAttribute.cs

```diff
@@ -45,24 +45,22 @@ public TAttribute Attribute
         /// <param name="modelMetadata">The <see cref="ModelMetadata"/> associated with the model annotated with
         /// <see cref="Attribute"/>.</param>
         /// <returns>Formatted error string.</returns>
-        protected virtual string GetErrorMessage(ModelMetadata modelMetadata)
+        protected virtual string GetErrorMessage(ModelMetadata modelMetadata, params object[] arguments)
         {
             if (modelMetadata == null)
             {
                 throw new ArgumentNullException(nameof(modelMetadata));
             }
 
-            var displayName = modelMetadata.GetDisplayName();
-
             if (_stringLocalizer != null &&
                 !string.IsNullOrEmpty(Attribute.ErrorMessage) &&
                 string.IsNullOrEmpty(Attribute.ErrorMessageResourceName) &&
                 Attribute.ErrorMessageResourceType == null)
```

Comment:
Need a couple tests of these conditions. Might already exist but I'm not seeing them. For example, demonstrate the `IStringLocalizer` isn't used if `ErrorMessageResourceName` or `ErrorMessageResourceType` is set. Best to create a test `ValidationAttributeAdapter<T>` subclass and exercise it a bit. May need a new test class.